src/Generators/EntityGenerator.php

```diff
@@ -230,17 +228,25 @@ protected function getRelatedModels(string $model, string $creatableClass): arra
                 continue;
             }
 
-            $relatedModels[] = class_basename(get_class($result->getRelated()));
+            $relatedModels[] = $this->removePath(get_class($result->getRelated()), $this->paths['models']);
         }
 
         DB::rollBack();
 
         return $relatedModels;
     }
 
+    protected function removePath(string $string, string $prefix): string
+    {
+        $normalizedString = str_replace('\\', '/', $string);
```

Comment:
let's implement separate method for it

```suggestion
        $normalizedString = $this->namespaceToPath($namespace);
```